### PR TITLE
Conversion of DRM selftests to KUnit tests

### PR DESCRIPTION
**Pull request by mairacanal**
As mentioned in the last meeting, I compiled all the tests developed during the LKCAMP hackathon and added a few more. The test-drm_mm and test-drm_buddy will take a little longer to convert and probably will be ready by the weekend.

I also need to write the cover letter for this patch set, but I decided to make it public as soon as possible so that we can gather some feedback and I can work on them during the week.

For those that are not familiar with the LKCAMP hackathon, in this email, Nicolas explains better what was the goal of this patchset: https://groups.google.com/g/kunit-dev/c/YqFR1q2uZvk/m/IbvItSfHBAAJ

The feedback gathered on the RFC (https://lore.kernel.org/all/20211111003453.209115-1-andrealmeid@collabora.com/T/) was already taken care of.

### CODE REVIEW

**Comment by magalilemes** [comment]
Nothing to point out, really, but one suggestion: I got very fond of the idea of having a separate Kconfig for the tests, like the one we have for the DML tests. So, what do you think about moving the config entries to `drm/tests/Kconfig`, and also having a general `DRM_KUNIT_TEST` entry for enabling all the tests at once?

**Comment by mairacanal** [comment]
@Grillo-0 if you like to set in and review the tests also, you are more than welcome.

**Comment by mairacanal** [comment]
> Nothing to point out, really, but one suggestion: I got very fond of the idea of having a separate Kconfig for the tests, like the one we have for the DML tests. So, what do you think about moving the config entries to `drm/tests/Kconfig`, and also having a general `DRM_KUNIT_TEST` entry for enabling all the tests at once?

@magalilemes Great idea! I'll work on it.

**Comment by mairacanal** [comment]
V1 -> V2:
- The suites not longer end in _tests

**Comment by mairacanal** [comment]
V1 -> V2:

- The suites not longer end in _tests
- Remove the TODO entry involving the conversion of selftests to KUnit

**Comment by mairacanal** [comment]
V1 -> V2:

- The suites not longer end in _tests
- Remove the TODO entry involving the conversion of selftests to KUnit
- Change the filenames to match the documentation: use *_test.c

File: drivers/gpu/drm/tests/Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: GPL-2.0-only
+obj-$(CONFIG_DRM_DAMAGE_HELPER_KUNIT_TEST) += test-drm_damage_helper.o
+obj-$(CONFIG_DRM_CMDLINE_PARSER_KUNIT_TEST) += test-drm_cmdline_parser.o
+obj-$(CONFIG_DRM_RECT_KUNIT_TEST) += test-drm_rect.o
+obj-$(CONFIG_DRM_FORMAT_KUNIT_TEST) += test-drm_format.o
+obj-$(CONFIG_DRM_PLANE_HELPER_KUNIT_TEST) += test-drm_plane_helper.o
+obj-$(CONFIG_DRM_DP_MST_HELPER_KUNIT_TEST) += test-drm_dp_mst_helper.o
+obj-$(CONFIG_DRM_FRAMEBUFFER_KUNIT_TEST) += test-drm_framebuffer.o
+obj-$(CONFIG_DRM_BUDDY_KUNIT_TEST) += test-drm_buddy.o
+obj-$(CONFIG_DRM_MM_KUNIT_TEST) += test-drm_mm.o
```
Comment (tales-aparecida) [inline drivers/gpu/drm/tests/Makefile]: I guess the linking paths were overlooked when the tests got renamed from `test-*.c` to `*_test.c`

File: drivers/gpu/drm/tests/drm_dp_mst_helper_test.c [hunk 2]
```diff
@@ -5,15 +5,15 @@
...
 #define PREFIX_STR "[drm_dp_mst_helper]"
 
+#include <kunit/test.h>
 #include <linux/random.h>
 
 #include <drm/display/drm_dp_mst_helper.h>
 #include <drm/drm_print.h>
 
 #include "../display/drm_dp_mst_topology_internal.h"
-#include "test-drm_modeset_common.h"
```
Comment (tales-aparecida) [inline drivers/gpu/drm/tests/drm_dp_mst_helper_test.c:14]: 🤔

Comment (mairacanal) [reply drivers/gpu/drm/tests/drm_dp_mst_helper_test.c:14]: This file is used to hold the signature of the tests. With KUnit, it becomes useless.

File: drivers/gpu/drm/tests/Makefile (hunk 1 above)
Comment (mairacanal) [reply drivers/gpu/drm/tests/Makefile]: My bad! I will fix (and also turn it into single entry as requested by Javier)

**Comment by mairacanal** [comment]
V1 -> V2:

- The suites not longer end in _tests
- Remove the TODO entry involving the conversion of selftests to KUnit
- Change the filenames to match the documentation: use *_test.c
- Add MODULE_LICENSE to all tests

File: drivers/gpu/drm/tests/Makefile (hunk 1 above)
Comment (mairacanal) [reply drivers/gpu/drm/tests/Makefile]: Fixed!

**Comment by mairacanal** [comment]
V1 -> V2:

- The suites not longer end in _tests
- Remove the TODO entry involving the conversion of selftests to KUnit
- Change the filenames to match the documentation: use *_test.c
- Add MODULE_LICENSE to all tests
- As already agreed, uses a generic symbol to group all tests (DRM_KUNIT_TEST)
- Add .kunitconfig on the first patch (it was on the second patch of the series)

File: drivers/gpu/drm/Makefile [hunk 3]
```diff
@@ -75,7 +75,7 @@ obj-$(CONFIG_DRM_KMS_HELPER) += drm_kms_helper.o
 # Drivers and the rest
 #
 
-obj-$(CONFIG_DRM_DEBUG_SELFTEST) += selftests/
+obj-y += tests/
```
Comment (tales-aparecida) [inline drivers/gpu/drm/Makefile]: just fyi, José uses 
`obj-$(CONFIG_DRM_KUNIT_TEST) += tests/`

**Comment by mairacanal** [comment]
V1 -> V2:

- The suites not longer end in _tests
- Remove the TODO entry involving the conversion of selftests to KUnit
- Change the filenames to match the documentation: use *_test.c
- Add MODULE_LICENSE to all tests
- As already agreed, uses a generic symbol to group all tests (DRM_KUNIT_TEST)
- Add .kunitconfig on the first patch (it was on the second patch of the series)
- Straightforward conversion of the drm_cmdline_parser tests without functional changes

**Comment by mairacanal** [comment]
v1 -> v2: https://lore.kernel.org/dri-devel/20220615135824.15522-1-maira.canal@usp.br/T/

- The suites not longer end in _tests (David Gow).
- Remove the TODO entry involving the conversion of selftests to KUnit (Javier Martinez Canillas).
- Change the filenames to match the documentation: use *_test.c (Javier Martinez Canillas).
- Add MODULE_LICENSE to all tests (kernel test robot).
- Make use of a generic symbol to group all tests - DRM_KUNIT_TEST (Javier Martinez Canillas).
- Add .kunitconfig on the first patch (it was on the second patch of the series).
- Straightforward conversion of the drm_cmdline_parser tests without functional changes (Shuah Khan)
- Add David's Tested-by tags

**Comment by mairacanal** [comment]
v1 -> v2: https://lore.kernel.org/dri-devel/20220615135824.15522-1-maira.canal@usp.br/T/

- The suites not longer end in _tests (David Gow).
- Remove the TODO entry involving the conversion of selftests to KUnit (Javier Martinez Canillas).
- Change the filenames to match the documentation: use *_test.c (Javier Martinez Canillas).
- Add MODULE_LICENSE to all tests (kernel test robot).
- Make use of a generic symbol to group all tests - DRM_KUNIT_TEST (Javier Martinez Canillas).
- Add .kunitconfig on the first patch (it was on the second patch of the series).
- Straightforward conversion of the drm_cmdline_parser tests without functional changes (Shuah Khan)
- Add David's Tested-by tags

v2 -> v3: https://lore.kernel.org/dri-devel/20220621200926.257002-1-maira.canal@usp.br/

- Change KUNIT_EXPECT_FALSE to KUNIT_EXPECT_EQ on drm_format_test (Daniel Latypov)
- Add Daniel's Acked-by tag

**Comment by mairacanal** [comment]
v1 -> v2: https://lore.kernel.org/dri-devel/20220615135824.15522-1-maira.canal@usp.br/T/

- The suites not longer end in _tests (David Gow).
- Remove the TODO entry involving the conversion of selftests to KUnit (Javier Martinez Canillas).
- Change the filenames to match the documentation: use *_test.c (Javier Martinez Canillas).
- Add MODULE_LICENSE to all tests (kernel test robot).
- Make use of a generic symbol to group all tests - DRM_KUNIT_TEST (Javier Martinez Canillas).
- Add .kunitconfig on the first patch (it was on the second patch of the series).
- Straightforward conversion of the drm_cmdline_parser tests without functional changes (Shuah Khan)
- Add David's Tested-by tags.

v2 -> v3: https://lore.kernel.org/dri-devel/20220621200926.257002-1-maira.canal@usp.br/

- Rebase it on top of the drm-misc-next with drm_format_helper KUnit tests.
- Change KUNIT_EXPECT_FALSE to KUNIT_EXPECT_EQ on drm_format_test (Daniel Latypov).
- Add Daniel's Acked-by tag.

**Comment by mairacanal** [comment]
v1 -> v2: https://lore.kernel.org/dri-devel/20220615135824.15522-1-maira.canal@usp.br/T/

- The suites not longer end in _tests (David Gow).
- Remove the TODO entry involving the conversion of selftests to KUnit (Javier Martinez Canillas).
- Change the filenames to match the documentation: use *_test.c (Javier Martinez Canillas).
- Add MODULE_LICENSE to all tests (kernel test robot).
- Make use of a generic symbol to group all tests - DRM_KUNIT_TEST (Javier Martinez Canillas).
- Add .kunitconfig on the first patch (it was on the second patch of the series).
- Straightforward conversion of the drm_cmdline_parser tests without functional changes (Shuah Khan)
- Add David's Tested-by tags.

v2 -> v3: https://lore.kernel.org/dri-devel/20220621200926.257002-1-maira.canal@usp.br/

- Rebase it on top of the drm-misc-next with drm_format_helper KUnit tests.
- Change KUNIT_EXPECT_FALSE to KUNIT_EXPECT_EQ on drm_format_test (Daniel Latypov).
- Add Daniel's Acked-by tag.

v3 -> v4: https://lore.kernel.org/dri-devel/9185aadb-e459-00fe-70be-3675f6f3ef4c@redhat.com/T/

- Add blank line after #include <kunit/test.h>  (Javier Martinez Canillas).
- Make the order of the tags chronological  (Javier Martinez Canillas).
- Add Copywrite to the tests (Javier Martinez Canillas).
- Add  Javier Martinez Canillas's Reviewed-By tag.

**Comment by mairacanal** [comment]
v1 -> v2: https://lore.kernel.org/dri-devel/20220615135824.15522-1-maira.canal@usp.br/T/

- The suites not longer end in _tests (David Gow).
- Remove the TODO entry involving the conversion of selftests to KUnit (Javier Martinez Canillas).
- Change the filenames to match the documentation: use *_test.c (Javier Martinez Canillas).
- Add MODULE_LICENSE to all tests (kernel test robot).
- Make use of a generic symbol to group all tests - DRM_KUNIT_TEST (Javier Martinez Canillas).
- Add .kunitconfig on the first patch (it was on the second patch of the series).
- Straightforward conversion of the drm_cmdline_parser tests without functional changes (Shuah Khan)
- Add David's Tested-by tags.

v2 -> v3: https://lore.kernel.org/dri-devel/20220621200926.257002-1-maira.canal@usp.br/

- Rebase it on top of the drm-misc-next with drm_format_helper KUnit tests.
- Change KUNIT_EXPECT_FALSE to KUNIT_EXPECT_EQ on drm_format_test (Daniel Latypov).
- Add Daniel's Acked-by tag.

v3 -> v4: https://lore.kernel.org/dri-devel/9185aadb-e459-00fe-70be-3675f6f3ef4c@redhat.com/T/

- Add blank line after #include <kunit/test.h>  (Javier Martinez Canillas).
- Make the order of the tags chronological  (Javier Martinez Canillas).
- Add Copywrite to the tests (Javier Martinez Canillas).
- Add  Javier Martinez Canillas's Reviewed-By tag.

v4 -> v5: https://lore.kernel.org/dri-devel/20220702131116.457444-1-maira.canal@usp.br/T/

- Fix checkpatch warnings and checks (Javier Martinez Canillas).

**Pull request closed without merging**